### PR TITLE
Add sccache to CI tests

### PR DESCRIPTION
**Pull request by zanieb**

### CODE REVIEW

**Comment by MichaReiser** [comment]
Where does sccache store the cache artifacts?

**Comment by zanieb** [comment]
I think it reuses the GitHub Actions cache. It doesn't appear to be faster here though.

**Comment by MichaReiser** [comment]
Screenshot 2023-10-19 at 09 08 18

The caches are much smaller...

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -96,7 +100,10 @@ jobs:
         uses: taiki-e/install-action@v2
         with:
           tool: cargo-insta
-      - uses: Swatinem/rust-cache@v2
+      - name: Set up basic Rust cache
+        uses: Swatinem/rust-cache@v2
```
Comment (MichaReiser) [inline .github/workflows/ci.yaml:104]: Should we remove `rust-cache` for the test to avoid double caching (or the two conflicting)?

Comment (zanieb) [reply .github/workflows/ci.yaml:104]: The pull request we were referencing (that showed a speedup) had both setup

**Comment by zanieb** [comment]
If it's smaller overall that'd probably be good, I don't think things got slower..

**Pull request closed without merging**